src/components/src/test/java/org/apache/jmeter/gui/action/template/TestTemplateManager.java

```diff
@@ -72,30 +82,60 @@ public void testTemplateFile() throws IOException, SAXException, ParserConfigura
         assertEquals("/bin/templates/testTemplate.jmx", testTemplate.getFileName());
         assertEquals("testTemplate desc", testTemplate.getDescription());
         assertNull(testTemplate.getParameters());
-
     }
 
-    /**
-     * Check that a wrong xml file throws a FileNotFoundException
-     */
     @Test(expected = FileNotFoundException.class)
-    public void testInvalidTemplateFile() throws Exception {
-        String xmlTemplatePath = "missing.xml";
-        File templateFile = new File(xmlTemplatePath);
-        TemplateManager templateManager = TemplateManager.getInstance();
-        templateManager.parseTemplateFile(templateFile);
+    public void testNonExistantXmlFileThrowsFileNotFoundException() throws Exception {
+        File xmlTemplateFile = new File("missing.xml");
+        TemplateManager.getInstance().parseTemplateFile(xmlTemplateFile);
     }
 
     @Test
     public void testInvalidTemplateXml() throws IOException, SAXException, ParserConfigurationException {
         try {
             String xmlTemplatePath = this.getClass().getResource("invalidTemplates.xml").getFile();
             File templateFile = new File(xmlTemplatePath);
-            TemplateManager templateManager = TemplateManager.getInstance();
-            templateManager.parseTemplateFile(templateFile);
+            TemplateManager.getInstance().parseTemplateFile(templateFile);
         } catch (SAXParseException ex) {
-            assertTrue("Exception did not contains expected message, got:"+ex.getMessage(),
-                    ex.getMessage().indexOf("Element type \"key\" must be declared.")>=0);
+            assertTrue("Exception did not contains expected message, got:" + ex.getMessage(),
+                    ex.getMessage().contains("Element type \"key\" must be declared."));
         }
     }
+
+    @Test
+    public void testDifferentTemplatesAreNotEqual() {
+        Template testTemplate1 = templateMap.get("testTemplateWithParameters");
+        Template testTemplate2 = templateMap.get("testTemplateNotTestPlan");
+        Template testTemplate3 = templateMap.get("testTemplate");
+
+        assertNotEquals(testTemplate1, testTemplate2);
+        assertNotEquals(testTemplate1, testTemplate3);
+        assertNotEquals(testTemplate2, testTemplate1);
+        assertNotEquals(testTemplate2, testTemplate3);
+        assertNotEquals(testTemplate3, testTemplate1);
+        assertNotEquals(testTemplate3, testTemplate2);
+
+        assertNotEquals(testTemplate1.hashCode(), testTemplate2.hashCode());
```

Comment:
This is inconsistent. It would result in flaky tests in case of hashcode collision.

Have you seen https://jqno.nl/equalsverifier/ ?